src/main/java/com/cs203/smucode/controllers/UserController.java

```diff
@@ -95,6 +95,34 @@ public ResponseEntity<String> updateRating(@RequestBody @Valid UserRatingDTO rat
         }
     }
 
+    @PutMapping("/update-win/{username}")
+    public ResponseEntity<String> updateWin(@PathVariable String username) {
+        try {
+            validateUsername(username);
+            userService.updateUserWin(username);
+
+            return ResponseEntity.ok("User win count updated successfully");
+        } catch (ApiRequestException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new ApiRequestException("An error occurred while updating the user's win count");
+        }
+    }
+
+    @PutMapping("/update-loss/{username}")
+    public ResponseEntity<String> updateLoss(@PathVariable String username) {
+        try {
+            validateUsername(username);
+            userService.updateUserLoss(username);
+
+            return ResponseEntity.ok("User loss count updated successfully");
+        } catch (ApiRequestException e) {
+            throw e;
```

Comment:
dont throw e, throw the message as u are throwing the entire stakk trace (remember what adrian mention_)